### ISSUE TITLE
Flag to ignore suffix-less @SuppressWarnings("all"), or to require both prefix&suffix in suppressed keys

### ISSUE THREAD

**Issue by PawelLipski** [typetools/checker-framework#3356]
A mechanism to do either of the above would be pretty useful. Certain tools (like Lombok) generate `@SuppressWarnings("all")` on generated methods. (Lombok doesn't even allow to switch this generating this suppression off, unless `delombok` is being run... which in turn has its own host of problems, like completely mangling line numbers in compilation errors/runtime stack traces). This leads to false negatives under certain circumstances, let's e.g consider the following interface:

```java
interface IFoo {
  String getFoo();
}
```

and an implementing class:

```java
import org.checkerframework.checker.nullness.qual.Nullable;

class Foo implements IFoo {
  @lombok.Getter
  @Nullable // overridden getter must NOT be nullable since it's declared non-null in IFoo
  private final String foo;
}
```

The Lombok-generated getter will be something like:

```java
  @org.checkerframework.checker.nullness.qual.Nullable
  @SuppressWarnings("all")
  @lombok.Generated
  public String getFoo() {
    return this.foo;
  }
```

**If it wasn't for `@SuppressWarnings("all")`, then `NullnessChecker` would issue an error, which it definitely should in this case**: a method declared as returning non-null in the interface can't be implemented as possibly returning null.

Either a flag to ignore suffix-less `all` key in `@SuppressWarnings`, or a flag similar to `-ArequirePrefixInWarningSuppressions` (maybe `-ArequirePrefixAndSuffixInWarningSuppressions`?) would alleviate the problem.

**Comment by smillst** [comment]
(See related https://github.com/rzwitserloot/lombok/issues/2433.)

**Comment by mernst** [comment]
Currently, `@SuppressWarnings("all")` is ambiguous to the Checker Framework.
 * It could mean "all warnings", like `@SuppressWarnings({"assignment.type.incompatible", "group.count.invalid", ...})`
 * It could mean "all type systems", like `@SuppressWarnings({"nullness", "index", ...})`

By default, there is so difference between the two:  each one suppresses the same warnings (namely, all warnings!).

However, when `-ArequirePrefixInWarningSuppressions` is supplied, the Checker Framework needs to distinguish the two senses of `@SuppressWarnings("all")`.  The former is an error (it has no prefix), but the latter is acceptable.

We will change the syntax of warning suppressions so that "all" is no longer a prefix, but "allcheckers" is.
